android/app/src/main/java/org/fossasia/openevent/fragments/SpeakersListFragment.java

```diff
@@ -172,9 +172,28 @@ public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.action_sort:
 
+                //checking for org and country
+                int[] array = DbSingleton.getInstance().getCount().get(0);
```

Comment:
Better :clap: 

But here's the issue, it gets called every time the user clicks on menu and on the Main Thread!

Furthermore, you already have a list of speakers, why querying it again?

Instead of querying it again, calculate it in the subscribe method, where the speakers are loaded and set the corresponding state variables and then just check them while loading menu, like you have done here